Tasks/DeployVisualStudioTestAgent/TestAgentInstall.ps1

```diff
@@ -3,13 +3,24 @@ function Install-Product($SetupPath, $UserName, $Password, $ProductVersion, $Arg
 	$InstalledCheckRegKey = ("SOFTWARE\Microsoft\DevDiv\vstf\Servicing\{0}\testagentcore" -f $ProductVersion)
 	$InstalledCheckRegValueName = "Install"
 	$InstalledCheckRegValueData = "1"
-
-	$isProductExists = Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName $InstalledCheckRegValueName -InstalledCheckRegValueData $InstalledCheckRegValueData
-        $testAgentFileExists = Test-Path "$env:SystemDrive\TestAgent\testagent"
-
-	if($testAgentFileExists -and $isProductExists)
+                
+	$isProductExists = Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName $InstalledCheckRegValueName         
+
+        $versionToInstall = ((Get-Item $SetupPath).VersionInfo.FileVersion) 
+        $versionInstalled = (Get-ProductEntry -InstalledCheckRegKey $InstalledCheckRegKey -InstalledCheckRegValueName "version")
+
+        if($versionToInstall -ne $null)
```

Comment:
instead of doing this why not use properties that version info returns. just contruct the file version object and use the properties.